### PR TITLE
Implementar botão de compra

### PR DESCRIPTION
**Pull request by markx3**
Well I guess I went overboard.

Sorry.

### CODE REVIEW

File: components/confirmation/Confirmation.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  }
+
+  render() {
+    const { navigation } = this.props
+
+    const userName = navigation.getParam('userName')
+    const productName = navigation.getParam('productName')
+    const productPrice = navigation.getParam('productPrice')
+
+    const handleConfirm = async () => {
```
Comment (izn) [inline components/confirmation/Confirmation.js]: Should be moved outside `render`. Too much complexity to be stored into a weak variable. WDYT?

File: components/confirmation/Confirmation.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            product: productName,
+            price: productPrice,
+          }
+        })
+
+        Alert.alert(`Compra efetuada.\nObrigado, ${userName}! 🥳`)
+      } catch (error) {
+        Alert.alert(`Ocorreu um erro ao confirmar a compra. 🤕`)
+
+        console.error(error)
```
Comment (izn) [inline components/confirmation/Confirmation.js]: Is this useful?

File: helpers/actions.js [hunk 3]
```diff
@@ -23,3 +23,15 @@ export const fetchProducts = async () => {
     return { products: [] }
   }
 }
+
+export const createTransaction = async ({ payload }) => {
```
Comment (izn) [inline helpers/actions.js]: I'm **REALLY** SORRY.

```es6
export const createTransaction = async ({ payload }) => await post({ action: 'create_transaction', payload })
```

I think we don't need to use a try/catch here.

File: components/confirmation/Confirmation.js (hunk 1 above)
Comment (markx3) [reply components/confirmation/Confirmation.js]: I honestly didn't see it was inside `render`.

Need some sleep yo. 💯

File: components/confirmation/Confirmation.js (hunk 2 above)
Comment (markx3) [reply components/confirmation/Confirmation.js]: Not really, no.

Comment (izn) [reply components/confirmation/Confirmation.js]: Sorry.

File: components/confirmation/Confirmation.js (hunk 1 above)
Comment (izn) [reply components/confirmation/Confirmation.js]: Sorry.

File: helpers/actions.js (hunk 3 above)
Comment (izn) [reply helpers/actions.js]: Also, I'm not sure if we need `await` in this context.

Comment (markx3) [reply helpers/actions.js]: We don't use try/catch here, just `await` 🤔 💭

Comment (izn) [reply helpers/actions.js]: [Are you sure?](https://github.com/izn/flipermercado/pull/2/files/5919cb698d998fc7627c66aaf7ec935338115f22#diff-a2fe2f10eeae01b8e7e1bac5a4162898R28)

Comment (markx3) [reply helpers/actions.js]: I'm not sure.

Comment (markx3) [reply helpers/actions.js]: Actually we may need this. If we don't return a `Promise` here we can't `await` it on the component. :/

I'm thinking of adding a `throw new Error(error)` to the `catch` statement so we are able to catch it on the component.

🌴 🏖 What do you think?

Comment (markx3) [reply helpers/actions.js]: But throwing on a catch is weird, isn't it?

Comment (markx3) [reply helpers/actions.js]: `throw on a catch` ~> catchy band name

**Pull request merged**